tester/reporter.go

```diff
@@ -142,7 +143,17 @@ func (r JSONCoverageReporter) Report(ch chan *Result) error {
 	report := r.Cover.Report(r.Modules)
 	encoder := json.NewEncoder(r.Output)
 	encoder.SetIndent("", "  ")
-	return encoder.Encode(report)
+	if err := encoder.Encode(report); err != nil {
```

Comment:
I was a bit surprised to see both the coverage report _and_ the error message when the threshold is not met. Let's move the coverage check before the report is emitted. That way if the coverage check fails, the report will not be printed.